src/metabase/notification/payload/core.clj

```diff
@@ -20,43 +22,115 @@
+        [:parameters {:optional true} [:maybe [:sequential :map]]]
+        [:dashboard_subscription_dashcards {:optional true}
+         [:sequential [:map
+                       [:card_id                        [:maybe ms/PositiveInt]]
+                       [:include_csv   {:optional true} [:maybe ms/BooleanValue]]
+                       [:include_xls   {:optional true} [:maybe ms/BooleanValue]]
+                       [:format_rows   {:optional true} [:maybe ms/BooleanValue]]
+                       [:pivot_results {:optional true} [:maybe ms/BooleanValue]]]]]]]]]
     ;; for testing only
-    [:notification/testing       :map]]])
+    [:notification/testing :map]]])
 
 (def NotificationPayload
   "Schema for the notification payload."
   ;; TODO: how do we make this schema closed after :merge?
   [:merge
-   Notification
    [:map
-    [:context [:map]]]
+    [:payload_type (into [:enum] models.notification/notification-types)]
+    [:context      [:map]]]
    [:multi {:dispatch :payload_type}
     ;; override payload to add extra-context key
-    [:notification/system-event [:map
-                                 ;; override the payload with extra context
-                                 [:payload
-                                  [:map {:closed true}
-                                   [:event_topic                   [:fn #(= "event" (-> % keyword namespace))]]
-                                   [:event_info                    [:maybe :map]]
-                                   [:custom       {:optional true} [:maybe :map]]]]]]
-    [:notification/testing       :map]]])
+    [:notification/system-event
+     [:map
+      ;; override the payload with extra context
+      [:payload
+       [:map {:closed true}
+        [:event_topic                   [:fn #(= "event" (-> % keyword namespace))]]
+        [:event_info                    [:maybe :map]]
+        [:custom       {:optional true} [:maybe :map]]]]]]
+    [:notification/dashboard-subscription
+     [:map
+      [:payload [:map
+                 [:dashboard_parts             [:sequential notification.payload.execute/Part]]
+                 [:dashboard                   :map]
+                 [:dashboard_subscription      :map]
+                 [:style                       :map]
+                 [:parameters {:optional true} [:maybe [:sequential :map]]]]]]]
+    [:notification/alert
+     [:map
+      [:payload [:map
+                 [:card_part [:maybe notification.payload.execute/Part]]
+                 [:card      :map]
+                 [:style     :map]
+                 [:alert     :map]]]]]
+    [:notification/testing   :map]]])
+
+;; TODO: from metabase.email.messages
+(defn logo-url
+  "Return the URL for the application logo. If the logo is the default, return a URL to the Metabase logo."
+  []
+  (let [url (public-settings/application-logo-url)]
+    (cond
+      (= url "app/assets/img/logo.svg") "http://static.metabase.com/email_logo.png"
+
+      :else nil)))
```

Comment:
I missed the NOTE where we temporarily disable a check, having that make this `cond-worthy` I think :D

```clojure
(defn logo-url
  "Return the URL for the application logo. If the logo is the default, return a URL to the Metabase logo."
  []
  (let [url (public-settings/application-logo-url)]
    (cond
      (= url "app/assets/img/logo.svg") "http://static.metabase.com/email_logo.png"
      ;; NOTE: disabling whitelabeled URLs for now since some email clients don't render them correctly
      ;; We need to extract them and embed as attachments like we do in metabase.pulse.render.image-bundle
      ;; (data-uri-svg? url)               (themed-image-url url color)
      :else nil)))
```